platform/src/Playground.js

```diff
@@ -14,8 +16,9 @@ import { ToolManager as ToolsManager } from './ToolsManager.js';
 import { ConsolePanel } from "./ConsolePanel.js";
 import { ProgramPanel } from "./ProgramPanel.js";
 import { OutputPanel } from "./OutputPanel.js";
-import { TestPanel } from './TestPanel .js';
+import { TestPanel } from './TestPanel.js';
 import { BlankPanel } from './BlankPanel .js';
+import { XtextEditorPanel } from './XtextEditorPanel.js';
```

Comment:
This feels quite specialised. Do we have to have an explicit reference to Xtext in the main platform code?

Note, I'm not asking for this to be changed before we merge the code, but we should consider whether we can further refactor at some point.

Reply:
Yes there is work to be done generalise the Xtext workbench, I've added it as a task to the tracking issue #81. 

To use the generated editor that is an XtextServelet the education platform uses xtext-ace.js (provided by xtext) to handle and communicate changes to the editor instance. This is mostly because the platform uses the ace editor so it minimsed changes required to the platform. 

However, the LSP seems to be more widely supported and Xtext can also generate a LSP for languages created using it. The ace editor doesn't support LSP, however, there is support for it in the monaco editor/eclipse theia. Therefore, we should consider switching editors and using LSP for communication between generated editors.